### ISSUE TITLE
Error "'node._relayoutBoundary == _relayoutBoundary': is not true."

### ISSUE THREAD

**Issue by averyyan** [flutter/flutter#22557]
```
flutter: inputnull
flutter: inputnull
flutter: inputnull
flutter: inputnull
flutter: inputTextEditingController#c2b37(TextEditingValue(text: ┤├, selection: TextSelection(baseOffset: -1, extentOffset: -1, affinity: TextAffinity.downstream, isDirectional: false), composing: TextRange(start: -1, end: -1)))
flutter: ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
flutter: The following assertion was thrown building _ScrollableScope:
flutter: 'package:flutter/src/rendering/object.dart': Failed assertion: line 1358 pos 14:
flutter: 'node._relayoutBoundary == _relayoutBoundary': is not true.
flutter:
flutter: Either the assertion indicates an error in the framework itself, or we should provide substantially
flutter: more information in this error message to help you determine and fix the underlying cause.
flutter: In either case, please report this assertion by filing a bug on GitHub:
flutter:   https://github.com/flutter/flutter/issues/new
flutter:
flutter: When the exception was thrown, this was the stack:
flutter: #2      RenderObject._debugSubtreeRelayoutRootAlreadyMarkedNeedsLayout (package:flutter/src/rendering/object.dart:1358:14)
flutter: #3      RenderObject.markNeedsLayout (package:flutter/src/rendering/object.dart:1408:14)
flutter: #4      RenderBox.markNeedsLayout (package:flutter/src/rendering/box.dart:1842:11)
flutter: #5      RenderObject.markParentNeedsLayout (package:flutter/src/rendering/object.dart:1442:14)
flutter: #6      RenderBox.markNeedsLayout (package:flutter/src/rendering/box.dart:1838:9)
flutter: #7      RenderObject.markParentNeedsLayout (package:flutter/src/rendering/object.dart:1442:14)
flutter: #8      RenderBox.markNeedsLayout (package:flutter/src/rendering/box.dart:1838:9)
flutter: #9      RenderObject.markParentNeedsLayout (package:flutter/src/rendering/object.dart:1442:14)
flutter: #10     RenderBox.markNeedsLayout (package:flutter/src/rendering/box.dart:1838:9)
flutter: #11     RenderObject.markParentNeedsLayout (package:flutter/src/rendering/object.dart:1442:14)
flutter: #12     RenderBox.markNeedsLayout (package:flutter/src/rendering/box.dart:1838:9)
flutter: #13     RenderObject.markParentNeedsLayout (package:flutter/src/rendering/object.dart:1442:14)
flutter: #14     RenderBox.markNeedsLayout (package:flutter/src/rendering/box.dart:1838:9)
flutter: #15     RenderObject.markParentNeedsLayout (package:flutter/src/rendering/object.dart:1442:14)
flutter: #16     RenderBox.markNeedsLayout (package:flutter/src/rendering/box.dart:1838:9)
flutter: #17     RenderObject.markParentNeedsLayout (package:flutter/src/rendering/object.dart:1442:14)
flutter: #18     RenderBox.markNeedsLayout (package:flutter/src/rendering/box.dart:1838:9)
flutter: #19     RenderEditable.markNeedsTextLayout (package:flutter/src/rendering/editable.dart:483:5)
flutter: #20     RenderEditable.text= (package:flutter/src/rendering/editable.dart:493:5)
flutter: #21     _Editable.updateRenderObject (package:flutter/src/widgets/editable_text.dart:1025:9)
flutter: #22     RenderObjectElement.update (package:flutter/src/widgets/framework.dart:4430:12)
flutter: #23     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #24     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4803:14)
flutter: #25     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #26     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4803:14)
flutter: #27     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #28     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3689:16)
flutter: #29     Element.rebuild (package:flutter/src/widgets/framework.dart:3531:5)
flutter: #30     ProxyElement.update (package:flutter/src/widgets/framework.dart:3947:5)
flutter: #31     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #32     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4803:14)
flutter: #33     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #34     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4803:14)
flutter: #35     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #36     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4803:14)
flutter: #37     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #38     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3689:16)
flutter: #39     Element.rebuild (package:flutter/src/widgets/framework.dart:3531:5)
flutter: #40     StatefulElement.update (package:flutter/src/widgets/framework.dart:3835:5)
flutter: #41     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #42     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3689:16)
flutter: #43     Element.rebuild (package:flutter/src/widgets/framework.dart:3531:5)
flutter: #44     StatefulElement.update (package:flutter/src/widgets/framework.dart:3835:5)
flutter: #45     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #46     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3689:16)
flutter: #47     Element.rebuild (package:flutter/src/widgets/framework.dart:3531:5)
flutter: #48     StatefulElement.update (package:flutter/src/widgets/framework.dart:3835:5)
flutter: #49     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #50     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4803:14)
flutter: #51     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #52     _RenderDecorationElement._updateChild (package:flutter/src/material/input_decorator.dart:1255:30)
flutter: #53     _RenderDecorationElement.update (package:flutter/src/material/input_decorator.dart:1271:5)
flutter: #54     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #55     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3689:16)
flutter: #56     Element.rebuild (package:flutter/src/widgets/framework.dart:3531:5)
flutter: #57     StatefulElement.update (package:flutter/src/widgets/framework.dart:3835:5)
flutter: #58     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #59     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3689:16)
flutter: #60     Element.rebuild (package:flutter/src/widgets/framework.dart:3531:5)
flutter: #61     StatefulElement.update (package:flutter/src/widgets/framework.dart:3835:5)
flutter: #62     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #63     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4803:14)
flutter: #64     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #65     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3689:16)
flutter: #66     Element.rebuild (package:flutter/src/widgets/framework.dart:3531:5)
flutter: #67     StatefulElement.update (package:flutter/src/widgets/framework.dart:3835:5)
flutter: #68     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #69     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3689:16)
flutter: #70     Element.rebuild (package:flutter/src/widgets/framework.dart:3531:5)
flutter: #71     StatelessElement.update (package:flutter/src/widgets/framework.dart:3738:5)
flutter: #72     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #73     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4803:14)
flutter: #74     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #75     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4803:14)
flutter: #76     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #77     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3689:16)
flutter: #78     Element.rebuild (package:flutter/src/widgets/framework.dart:3531:5)
flutter: #79     StatefulElement.update (package:flutter/src/widgets/framework.dart:3835:5)
flutter: #80     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #81     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4803:14)
flutter: #82     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #83     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4803:14)
flutter: #84     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #85     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4803:14)
flutter: #86     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #87     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4803:14)
flutter: #88     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #89     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3689:16)
flutter: #90     Element.rebuild (package:flutter/src/widgets/framework.dart:3531:5)
flutter: #91     StatelessElement.update (package:flutter/src/widgets/framework.dart:3738:5)
flutter: #92     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #93     RenderObjectElement.updateChildren (package:flutter/src/widgets/framework.dart:4521:32)
flutter: #94     MultiChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4911:17)
flutter: #95     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #96     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4803:14)
flutter: #97     Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #98     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3689:16)
flutter: #99     Element.rebuild (package:flutter/src/widgets/framework.dart:3531:5)
flutter: #100    StatelessElement.update (package:flutter/src/widgets/framework.dart:3738:5)
flutter: #101    Element.updateChild (package:flutter/src/widgets/framework.dart:2729:15)
flutter: #102    _LayoutBuilderElement._layout.<anonymous closure> (package:flutter/src/widgets/layout_builder.dart:118:18)
flutter: #103    BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2253:19)
flutter: #104    _LayoutBuilderElement._layout (package:flutter/src/widgets/layout_builder.dart:107:11)
flutter: #105    RenderObject.invokeLayoutCallback.<anonymous closure> (package:flutter/src/rendering/object.dart:1727:58)
flutter: #106    PipelineOwner._enableMutationsToDirtySubtrees (package:flutter/src/rendering/object.dart:796:15)
flutter: #107    RenderObject.invokeLayoutCallback (package:flutter/src/rendering/object.dart:1727:13)
flutter: #108    _RenderLayoutBuilder.performLayout (package:flutter/src/widgets/layout_builder.dart:205:5)
flutter: #109    RenderObject.layout (package:flutter/src/rendering/object.dart:1631:7)
flutter: #110    RenderFlex.performLayout (package:flutter/src/rendering/flex.dart:738:15)
flutter: #111    RenderObject.layout (package:flutter/src/rendering/object.dart:1631:7)
flutter: #112    RenderPadding.performLayout (package:flutter/src/rendering/shifted_box.dart:199:11)
flutter: #113    RenderObject.layout (package:flutter/src/rendering/object.dart:1631:7)
flutter: #114    RenderPadding.performLayout (package:flutter/src/rendering/shifted_box.dart:199:11)
flutter: #115    RenderObject.layout (package:flutter/src/rendering/object.dart:1631:7)
flutter: #116    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:108:13)
flutter: #117    RenderObject.layout (package:flutter/src/rendering/object.dart:1631:7)
flutter: #118    RenderConstrainedBox.performLayout (package:flutter/src/rendering/proxy_box.dart:262:13)
flutter: #119    RenderObject.layout (package:flutter/src/rendering/object.dart:1631:7)
flutter: #120    RenderPadding.performLayout (package:flutter/src/rendering/shifted_box.dart:199:11)
flutter: #121    RenderObject.layout (package:flutter/src/rendering/object.dart:1631:7)
flutter: #122    RenderFlex.performLayout (package:flutter/src/rendering/flex.dart:738:15)
flutter: #123    RenderObject.layout (package:flutter/src/rendering/object.dart:1631:7)
flutter: #124    RenderOffstage.performLayout (package:flutter/src/rendering/proxy_box.dart:2882:14)
flutter: #125    RenderObject.layout (package:flutter/src/rendering/object.dart:1631:7)
flutter: #126    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:108:13)
flutter: #127    RenderObject.layout (package:flutter/src/rendering/object.dart:1631:7)
flutter: #128    RenderSliverList.performLayout (package:flutter/src/rendering/sliver_list.dart:164:27)
flutter: #129    RenderObject.layout (package:flutter/src/rendering/object.dart:1631:7)
flutter: #130    RenderSliverPadding.performLayout (package:flutter/src/rendering/sliver_padding.dart:182:11)
flutter: #131    RenderObject.layout (package:flutter/src/rendering/object.dart:1631:7)
flutter: #132    RenderViewportBase.layoutChildSequence (package:flutter/src/rendering/viewport.dart:405:13)
flutter: #133    RenderViewport._attemptLayout (package:flutter/src/rendering/viewport.dart:1320:12)
flutter: #134    RenderViewport.performLayout (package:flutter/src/rendering/viewport.dart:1238:20)
flutter: #135    RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:1506:7)
flutter: #136    PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:765:18)
flutter: #137    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:270:19)
flutter: #138    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding&WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:674:13)
flutter: #139    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:208:5)
flutter: #140    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:990:15)
flutter: #141    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:930:9)
flutter: #142    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:842:5)
flutter: #143    _invoke (dart:ui/hooks.dart:139:13)
flutter: #144    _drawFrame (dart:ui/hooks.dart:128:3)
flutter: (elided 2 frames from class _AssertionError)
flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════
flutter: Another exception was thrown: 'package:flutter/src/rendering/object.dart': Failed assertion: line 1358 pos 14: 'node._relayoutBoundary == _relayoutBoundary': is not true.
flutter: Another exception was thrown: 'package:flutter/src/foundation/node.dart': Failed assertion: line 106 pos 12: '_owner != null': is not true.
flutter: Another exception was thrown: 'package:flutter/src/rendering/object.dart': Failed assertion: line 1152 pos 12: 'child.parentData != null': is not true.
flutter: Another exception was thrown: 'package:flutter/src/rendering/object.dart': Failed assertion: line 1152 pos 12: 'child.parentData != null': is not true.
flutter: Another exception was thrown: 'package:flutter/src/foundation/node.dart': Failed assertion: line 106 pos 12: '_owner != null': is not true.
flutter: Another exception was thrown: 'package:flutter/src/rendering/object.dart': Failed assertion: line 1152 pos 12: 'child.parentData != null': is not true.
flutter: Another exception was thrown: 'package:flutter/src/foundation/node.dart': Failed assertion: line 106 pos 12: '_owner != null': is not true.
flutter: Another exception was thrown: 'package:flutter/src/rendering/object.dart': Failed assertion: line 1152 pos 12: 'child.parentData != null': is not true.
flutter: Another exception was thrown: 'package:flutter/src/foundation/node.dart': Failed assertion: line 106 pos 12: '_owner != null': is not true.
flutter: Another exception was thrown: 'package:flutter/src/rendering/object.dart': Failed assertion: line 1358 pos 14: 'node._relayoutBoundary == _relayoutBoundary': is not true.
flutter: Another exception was thrown: 'package:flutter/src/foundation/node.dart': Failed assertion: line 106 pos 12: '_owner != null': is not true.
flutter: Another exception was thrown: 'package:flutter/src/widgets/framework.dart': Failed assertion: line 2257 pos 20: '_debugCurrentBuildTarget == context': is not true.
```

**Comment by averyyan** [comment]
use TextField will make this

**Comment by Xavjer** [comment]
Do you mean `TextEditingController` ?

Please provide a sample of what you made that this error occured. I am using `TextEditingController` myself and have no issues with it.

Please write the details of the version you are using ( flutter --version )

**Comment by averyyan** [comment]
Flutter 0.9.5 • channel dev • https://github.com/flutter/flutter.git
Framework • revision 020fd590b0 (6 days ago) • 2018-09-26 14:28:26 -0700
Engine • revision 38a646e14c
Tools • Dart 2.1.0-dev.5.0.flutter-4cf2d3990b

import 'package:flutter/material.dart';

void main() => runApp(new MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return new MaterialApp(
      title: 'Flutter Demo',
      theme: new ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or press Run > Flutter Hot Reload in IntelliJ). Notice that the
        // counter didn't reset back to zero; the application is not restarted.
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  State<StatefulWidget> createState() {
    return _MyHomePage();
  }
}

class _MyHomePage extends State<MyHomePage> {
  final data = {};
  bool state = true;

  TextEditingController controller = TextEditingController();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("test"),
      ),
      body: Center(
        child: Column(
          children: <Widget>[
            GestureDetector(
              onTap: () {
                setState(() {
                  this.state = !this.state;
                  controller.clear();
                });
              },
              child: Center(child: Text("button")),
            ),
            Offstage(
              offstage: this.state,
              child: FormInput(
                label: 'test',
                data: this.data,
                keyword: 'test',
                controller: controller,
              ),
            ),
          ],
        ),
      ),
    );
  }
}

class FormInput extends StatelessWidget {
  final Map data;
  final String keyword;
  final String label;
  final double leftWidth;
  final String placeholder;

  final EdgeInsets padding;
  final int maxLines;
  final int maxLength;
  final TextInputType type;
  final bool obscureText;

  final TextEditingController controller;

  FormInput(
      {@required this.label,
      this.leftWidth = 45.0,
      this.padding = EdgeInsets.zero,
      this.placeholder,
      this.type,
      this.maxLines,
      this.maxLength,
      @required this.data,
      @required this.keyword,
      this.obscureText = false,
      this.controller});

  @override
  Widget build(BuildContext context) {
    return LayoutBuilder(
      builder: (BuildContext context, BoxConstraints constraints) {
        return Container(
          margin: EdgeInsets.only(bottom: 10.0, top: 10.0),
          child: Row(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: <Widget>[
              Container(
                padding: EdgeInsets.only(top: 6.0),
                width: this.leftWidth,
                child: Text(
                  this.label + "：",
                  style: TextStyle(
                    fontSize: 15.0,
                    fontWeight: FontWeight.bold,
                    color: Color.fromRGBO(0, 0, 0, 0.3),
                  ),
                ),
              ),
              Container(
                width: constraints.biggest.width - this.leftWidth,
                margin: EdgeInsets.all(0.0),
                decoration: BoxDecoration(
                    border: Border.all(
                        width: 1.0, color: Color.fromRGBO(231, 231, 231, 1.0)),
                    borderRadius: BorderRadius.circular(4.0)),
                child: TextField(
                    controller: this.controller,
                    obscureText: this.obscureText,
                    onChanged: (value) {
                      print(value);
                      this.data[this.keyword] = value.trim();
                    },
                    keyboardType: this.type,
                    maxLines: this.maxLines,
                    maxLength: this.maxLength,
                    decoration: InputDecoration(
                      contentPadding: EdgeInsets.all(5.0),
                      border: InputBorder.none,
                      hintText: this.placeholder,
                    )),
              ),
            ],
          ),
        );
      },
    );
  }
}

**Comment by zoechi** [comment]
Please explain the steps how to reproduce

**Comment by zoechi** [comment]
Ok, got it

- click button
- enter text
- click button again

**Comment by zoechi** [comment]
Looks similar to #5562

**Comment by zoechi** [comment]
#5562 is closed (can't reproduce) but this one still reproduces 
```
[✓] Flutter (Channel master, v1.2.2-pre.21, on Mac OS X 10.14.3 18D42, locale en-AT)
```

**Comment by Hixie** [comment]
Probably the same as https://github.com/flutter/flutter/issues/39151.

**Comment by goderbauer** [comment]
This doesn't reproduce anymore. I'd assume it was fixed with #39151.
